src/base/searchengine.h

```diff
@@ -38,10 +40,133 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
+{
+public:
+    typedef T ComponentType;
+    typedef Version<T,N> this_type;
+
+    Version()
+    {
+        m_components.fill(T());
+    }
+
+    template <class ...Other>
+    Version(Other... components)
+        : m_components {{components...}}
+    {
+    }
+
+    Version(const QString &version)
```

Comment:
They differ In the very first line  (`QLatin1Char` vs `char`). I don't know how to express this difference in a single template, otherwise I would join them.